From 576683faa9381cbba784eecd608f7ade10d3d356 Mon Sep 17 00:00:00 2001
From: Felix Fietkau <nbd@nbd.name>
Date: Wed, 22 Jul 2026 08:26:08 +0000
Subject: [PATCH 131/170] wifi: mt76: fix uninitialised RXDMAD_C descriptor
 info

[ Upstream commit d3ecac68f73b11828e72eaf7952a9beb5caea12b ]

Unlike other WED-RRO queues, RXDMAD_C frames continue into the skb build
path, but mt76_dma_get_buf() skips the desc->info read for RRO queues, so
the uninitialised on-stack info was stored into skb->cb and passed to
rx_skb(); initialise it to zero.

Fixes: e50d4d710efd ("wifi: mt76: Add mt76_dma_get_rxdmad_c_buf utility routione")
Link: https://patch.msgid.link/20260722082610.2699628-16-nbd@nbd.name
Signed-off-by: Felix Fietkau <nbd@nbd.name>
Signed-off-by: Sasha Levin <sashal@kernel.org>
---
 dma.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/dma.c
+++ b/dma.c
@@ -1002,7 +1002,7 @@ mt76_dma_rx_process(struct mt76_dev *dev
 
 	while (done < budget) {
 		bool drop = false;
-		u32 info;
+		u32 info = 0;
 
 		if (check_ddone) {
 			if (q->tail == dma_idx)
